lib/index.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+var version  = require( '../package.json').version;
+var url      = require('url');
+var request  = require( 'request');
+var _        = require( 'lodash' );
+
+//REST API Config Defaults
+var defaults = {
+  origin: 'https://api.sparkpost.com:443',
+  strictSSL: true,
+  apiVersion: 'v1'
+};
+
+var resolveUri = function( origin, uri ) {
+  if( !/^http/.test( uri ) ) {
+    uri = url.resolve( origin, uri );
+  }
+  return uri;
+};
+
+var SparkPost = function(apiKey, options) {
+
+  // only options passed in
+  if (typeof apiKey === 'object') {
+    options = apiKey;
+
+    this.apiKey = process.env.SPARKPOST_API_KEY;
+  } else {
+    options = options || {};
+
+    this.apiKey = apiKey || process.env.SPARKPOST_API_KEY;
+  }
+
+  if(typeof this.apiKey === 'undefined') {
+    throw new Error('Client requires an API Key.');
+  }
+
+  // adding version to object
+  this.version = version;
+
+  // setting up default headers
+  this.defaultHeaders = _.merge({
+    'User-Agent': 'node-sparkpost/' + this.version
+    , 'Content-Type': 'application/json'
+  }, options.headers);
+
+  //Optional client config
+  this.origin = options.origin || options.endpoint || defaults.origin;
+  this.strictSSL = typeof options.strictSSL !== 'undefined' ? options.strictSSL : defaults.strictSSL;
+  this.apiVersion = options.apiVersion || defaults.apiVersion;
+
+  this.sendingDomains = require('./sendingDomains')(this);
+  this.transmission = require('./transmission')(this);
+};
+
+SparkPost.prototype.request = function( options, callback ) {
+
+  // we need options
+  if( !_.isPlainObject( options ) ) {
+    throw new TypeError( 'options argument is required' );
+  }
+
+  var baseUrl = this.origin + '/api/' + this.apiVersion + '/';
+
+  // if we don't have a fully qualified URL let's make one
+  options.uri = resolveUri( baseUrl, options.uri );
+
+  // merge headers
+  options.headers = _.merge( {}, this.defaultHeaders, options.headers );
+
+  // add Authorization with API Key
+  options.headers.Authorization = this.apiKey;
+
+  // set Strict SSL
+  options.strictSSL = typeof options.strictSSL === 'undefined' ? this.strictSSL : options.strictSSL;
```

Comment:
I'm giving the option to override it in the request options. If you think that they should only have one chance to set it, we can do it, but I was trying to allow any request options to be set since this is pretty much a request wrapper.